sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -185,14 +186,18 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
   }
 
   private def isProbablyShuffleJoin(left: LogicalPlan,
-      right: LogicalPlan, hint: JoinHint): Boolean = {
-    !hintToBroadcastLeft(hint) && !hintToBroadcastRight(hint) &&
-      !canBroadcastBySize(left, conf) && !canBroadcastBySize(right, conf)
+                                    right: LogicalPlan,
+                                    hint: JoinHint,
+                                    joinType: JoinType): Boolean = {
+    // If any of the side can be broadcast, then it is not a shuffle join.
+    !canLeftSideBeBroadcast(left, conf, hint, joinType) &&
```

Comment:
I think you can simplify this since there are only two checks to
!(canLeftSideBeBroadcast(left, conf, hint, joinType) || canRightSideBeBroadcast(right, conf, hint, joinType))

Reply:
> I think you can simplify this since there are only two checks to !(canLeftSideBeBroadcast(left, conf, hint, joinType) || canRightSideBeBroadcast(right, conf, hint, joinType))

make sense. Modified it. Thanks.